src/clients/BundleDataClient/utils/FillUtils.ts

```diff
@@ -49,24 +50,31 @@ export function getRefundInformationFromFill(
 
 export function getRepaymentChainId(fill: Fill, matchedDeposit: Deposit): number {
   // Lite chain deposits force repayment on origin chain.
-  return matchedDeposit.fromLiteChain ? fill.originChainId : fill.repaymentChainId;
+  return matchedDeposit.fromLiteChain ? matchedDeposit.originChainId : fill.repaymentChainId;
 }
 
-export function isEvmRepaymentValid(
-  fill: Fill,
+export function willOverwriteRepaymentChain(
```

Comment:
nit: I'd probably prefer something like `forceDestinationRepayment` since `willOverwriteRepaymentChain` seems a bit ambiguous.

Reply:
i like that its more precise